lib/vagrant-lxc/action/private_networks.rb

```diff
@@ -26,16 +26,19 @@ def configure_private_networks(env)
             next if type.to_sym != :private_network
 
             container_name = env[:machine].provider.driver.container_name
+            address_type   = config[:type]
             ip             = config[:ip]
             bridge_ip      = config.fetch(:lxc__bridge_ip) { build_bridge_ip(ip) }
             bridge         = config.fetch(:lxc__bridge_name)
```

Comment:
IIRC this is kinda like the vagrant way of setting "type specific things". Like the `rsync__args` for the [rsync synced folder](https://docs.vagrantup.com/v2/synced-folders/rsync.html) or [`linux__nfs_options` for NFS synced folders](https://github.com/mitchellh/vagrant/blob/0cfe5ebfa62e2d4806617c7d828e36f693cf6752/plugins/hosts/linux/cap/nfs.rb#L110).

HTH